paymentsheet/src/test/java/com/stripe/android/customersheet/data/CustomerAdapterDataSourceTest.kt

```diff
@@ -0,0 +1,95 @@
+package com.stripe.android.customersheet.data
+
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.customersheet.CustomerAdapter
+import com.stripe.android.customersheet.ExperimentalCustomerSheetApi
+import com.stripe.android.customersheet.FakeCustomerAdapter
+import com.stripe.android.paymentsheet.model.SavedSelection
+import com.stripe.android.testing.PaymentMethodFactory
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+
+@OptIn(ExperimentalCustomerSheetApi::class)
+class CustomerAdapterDataSourceTest {
+    @Test
+    fun `on retrieve payment methods, should completely successfully from adapter`() = runTest {
+        val paymentMethods = PaymentMethodFactory.cards(size = 6)
+        val dataSource = createCustomerAdapterDataSource(
+            adapter = FakeCustomerAdapter(
+                paymentMethods = CustomerAdapter.Result.success(paymentMethods),
+            ),
+        )
+
+        val successResult = dataSource.retrievePaymentMethods().asSuccess()
```

Comment:
if this isn't a success, what's the error message on this test? It might be better to add an assert that the result is a success before attempting to cast it

Reply:
Will update the tests to do so.